lib/dialogue-agent/conversation.ts

```diff
@@ -548,4 +570,48 @@ export default class Conversation extends events.EventEmitter {
             console.log('Almond sends link: '+ url);
         return this._addMessage({ type: MessageType.LINK, url, title });
     }
+
+    appendLog(turn : DialogueTurn) {
+        this._log.push(turn);
+    }
```

Comment:
You'll need to create multiple dialogues, not just one, and move to the next dialogue every time the dialogue loop resets the state. Otherwise the state tracked at runtime and the state tracked in the file (by repeated applications of computeNewState) will be different.

Reply:
Are you saying I should have a list of dialogueTurn lists, and save them into separate files? I don't quite follow: when the dialogue resets, `this._dialogueState` is reset, and it is used to compute the new state. what else do I need to do?

Reply:
You should flush the current list of DialogueTurn lists to the DialogueParser when the state resets, so it closes the current dialogue in the file and it starts a new dialogue.
That way the state that you write in the file matches the state that you have in memory.